File: app/src/main/java/com/remotearthsolutions/expensetracker/contracts/HomeFragmentContract.java
package com.remotearthsolutions.expensetracker.contracts;

import com.remotearthsolutions.expensetracker.databaseutils.models.CategoryModel;
import com.remotearthsolutions.expensetracker.entities.ExpeneChartData;

import java.util.List;

public interface HomeFragmentContract {
    interface View {

        void showCategories(List<CategoryModel> categories);

        void loadExpenseChart(List<ExpeneChartData> listOfCategoryWithAmount);
    }
}

File: app/src/main/java/com/remotearthsolutions/expensetracker/contracts/HomeFragmentContract.kt
package com.remotearthsolutions.expensetracker.contracts

import com.remotearthsolutions.expensetracker.databaseutils.models.CategoryModel
import com.remotearthsolutions.expensetracker.entities.ExpenseChartData

interface HomeFragmentContract {
    interface View {
        fun showCategories(categories: List<CategoryModel>?)
        fun loadExpenseChart(listOfCategoryWithAmount: List<ExpenseChartData>?)
    }
}